### PR TITLE
New frontend metrics tracked events

### LINKED ISSUE #9679: Frontend metrics for other operations
- [x] SQL changed.
  - [x] Builder.
  - [x] Dataset.
- [x] CartoCSS applied builder.
- [x] Style form modified in builder.
- [x] Remove unnecessary metrics, already sent from the backend.
  - [x] Published map
  - [x] Removed map
  - [x] Exported map

### PR DESCRIPTION
**Pull request by MariaCheca**
Fixes #9679

New trackers for:
- SQL changed.
  - Builder.
  - Dataset.
- CartoCSS applied builder.
- Style form modified in builder.

Remove unnecessary metrics:
  - Published map
  - Removed map
  - Exported map

CR @xavijam @gfiorav

### CODE REVIEW

File: lib/carto/tracking/events.rb [hunk 1]
```diff
@@ -173,6 +173,33 @@ class AnalysisEvent < Event
       class CreatedAnalysis < AnalysisEvent; end
       class ModifiedAnalysis < AnalysisEvent; end
       class DeletedAnalysis < AnalysisEvent; end
+
+      class AppliedSql < Event
+        include Carto::Tracking::Services::Segment
```
Comment (gfiorav) [inline lib/carto/tracking/events.rb]: Like we spoke offline:

We don't want this event to be reported to Segment. It should be reported to Hubspot, but the connector is not yet implemented. Just remove this include and I'll remember to add the hubspot include when it's implemented.

File: lib/carto/tracking/events.rb [hunk 2]
```diff
@@ -173,6 +173,33 @@ class AnalysisEvent < Event
...
+
+        include Carto::Tracking::Validators::Visualization::Writable
+        include Carto::Tracking::Validators::User
+
+        required_properties :user_id, :visualization_id
+      end
+
+      class AppliedCartocss < Event
+        include Carto::Tracking::Services::Segment
```
Comment (gfiorav) [inline lib/carto/tracking/events.rb]: same here

File: lib/carto/tracking/events.rb [hunk 3]
```diff
@@ -173,6 +173,33 @@ class AnalysisEvent < Event
...
+
+        include Carto::Tracking::Validators::Visualization::Writable
+        include Carto::Tracking::Validators::User
+
+        required_properties :user_id, :visualization_id
+      end
+
+      class ModifiedStyleForm < Event
+        include Carto::Tracking::Services::Segment
```
Comment (gfiorav) [inline lib/carto/tracking/events.rb]: and here!

File: lib/assets/javascripts/cartodb3/editor/style/style-content-view.js [hunk 4]
```diff
@@ -224,6 +225,8 @@ module.exports = CoreView.extend({
     this._styleModel.set('autogenerated', false);
     this._layerDefinitionModel.set('cartocss_custom', false);
     this._userActions.saveLayer(this._layerDefinitionModel);
+
+    MetricsTracker.track('Modified Style Form', this._layerDefinitionModel.attributes);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/style/style-content-view.js]: I would not include all attributes, only `layer_id` and `carto_css`.

File: lib/assets/javascripts/cartodb3/editor/style/style-view.js [hunk 5]
```diff
@@ -126,6 +127,8 @@ module.exports = CoreView.extend({
       this._userActions.saveLayer(this._layerDefinitionModel);
 
       CartoCSSNotifications.showSuccessNotification();
+
+      MetricsTracker.track('Applied Cartocss', this._layerDefinitionModel.attributes);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/style/style-view.js]: I would not include all attributes, only `layer_id` and `carto_css`.

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/data/data-view.js [hunk 6]
```diff
@@ -230,6 +231,8 @@ module.exports = CoreView.extend({
     });
 
     this._checkClearButton();
+
+    MetricsTracker.track('Applied sql', this._nodeModel.attributes);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/data/data-view.js]: `node_id` and `sql`.

File: lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-options-view.js [hunk 7]
```diff
@@ -264,7 +265,10 @@ module.exports = CoreView.extend({
     this._layerDefinitionModel.save({
       sql: content
     });
+
     this._checkClearButton();
+
+    MetricsTracker.track('Applied sql', this._tableModel.attributes);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-options-view.js]: `node_id` and `sql`.

**Comment by xavijam** [comment]
:es: when ready.

File: lib/carto/tracking/events.rb [hunk 8]
```diff
@@ -173,6 +173,27 @@ class AnalysisEvent < Event
       class CreatedAnalysis < AnalysisEvent; end
       class ModifiedAnalysis < AnalysisEvent; end
       class DeletedAnalysis < AnalysisEvent; end
+
+      class AppliedSql < Event
+        include Carto::Tracking::Validators::Visualization::Writable
```
Comment (gfiorav) [inline lib/carto/tracking/events.rb]: Did you remember to check wether this restriction holds? Have you tested it in org with shared vis? Same goes for the other events.

Comment (MariaCheca) [inline lib/carto/tracking/events.rb]: Nope, taking a look ASAP. Thanks!

Comment (MariaCheca) [inline lib/carto/tracking/events.rb]: I did, and it's fine for shared visualizations in organizations as long as the user has write permissions (i.e. another user in the same org shares a dataset with her/him).

**Pull request merged**